src/abstractdataframe/selection.jl

```diff
@@ -0,0 +1,357 @@
+normalize_selection(idx::AbstractIndex, sel) = idx[sel]
```

Comment:
Can you add a comment explaining what this function does exactly to ease reading?

BTW, maybe it would be simpler to always normalize to `incol => fun => outcol`, with `fun == nothing` for renaming? Just and idea.

Reply:
OK. I will add it. I have thought about making "rename" a "transform" with `nothing`, but it would just add a branch we anyway have at a lower level of code. It was cleaner for me to explicitly differentiate renaming from transforming on the top-level (in particular when we rename we have to handle `copycols` a bit differently than when we transform).

Reply:
OK. I will go for an internal, unexported, singleton `ColRename` type.